code/modules/projectiles/guns/flamer.dm

```diff
@@ -30,6 +30,40 @@
 	fire_delay = 4
 
 
+/obj/item/weapon/gun/flamer/mini_flamer
+	name = "mini flamethrower"
+	desc = "A weapon-mounted refillable flamethrower attachment.\nIt is designed for short bursts."
+	icon = 'icons/Marine/marine-weapons.dmi'
+	icon_state = "flamethrower"
+	
+	flags_gun_features = GUN_UNUSUAL_DESIGN|GUN_AMMO_COUNTER|GUN_WIELDED_FIRING_ONLY|GUN_WIELDED_STABLE_FIRING_ONLY|GUN_IS_ATTACHMENT|GUN_ATTACHMENT_FIRE_ONLY
+	w_class = WEIGHT_CLASS_BULKY
+	fire_delay = 2.5 SECONDS
+	fire_sound = 'sound/weapons/guns/fire/flamethrower3.ogg'
+
+
+	current_mag = /obj/item/ammo_magazine/flamer_tank/mini
+	slot = ATTACHMENT_SLOT_UNDER
+	attach_icon_state = "flamethrower_a"
+	attach_delay = 3 SECONDS
+	detach_delay = 3 SECONDS
+	pixel_shift_x = 15
+	pixel_shift_y = 18
+
+
+/obj/item/weapon/gun/flamer/mini_flamer/light_pilot(mob/user, mustlit)
```

Comment:
how can you light the pilot?

Reply:
light_pilot is called to add an overlay to the flamers, i overwrote it since the miniflamer has the flame in its sprite. Its a shitcode solution to shitcode and I dont want to refactor flamers yet